src/FSharp.Analyzers/StringAnalyzer.fs

```diff
@@ -14,15 +14,24 @@ let StringStartsWithCode = "GRA-STRING-002"
 [<Literal>]
 let StringIndexOfCode = "GRA-STRING-003"
 
-let (|StringExpr|_|) (e : FSharpExpr) =
-    if e.Type.ErasedType.BasicQualifiedName = "System.String" then
-        Some ()
+let tryGetFullName (e : FSharpExpr) =
+    if e.Type.ErasedType.HasTypeDefinition then
+        match e.Type.ErasedType.TypeDefinition.TryGetFullName () with
```

Comment:
This probably doesn't need a match statement if you are wrapping it again.

Reply:
uh, yeah. I was in a tunnel there.